@stellar/typescript-wallet-sdk-soroban/src/Helpers/scValByType.ts

```diff
@@ -0,0 +1,100 @@
+import { StrKey, scValToNative, xdr } from "@stellar/stellar-sdk";
+
+/**
+ * This function attempts to convert smart contract (complex) value types
+ * to common/simpler types like string, array, buffer, JSON string, etc.
+ *
+ * @param {xdr.ScVal} scVal  the smart contract (complex) value
+ *
+ * @returns {any}  the smart contract value converted to a common/simpler
+ *     value like string, array, buffer, JSON string, etc.
+ *
+ * @example
+ *   const accountAddress = xdr.ScVal.scvAddress(
+ *     xdr.ScAddress.scAddressTypeAccount(
+ *       xdr.PublicKey.publicKeyTypeEd25519(
+ *         StrKey.decodeEd25519PublicKey("GBBM6BKZPEHWYO3E3YKREDPQXMS4VK35YLNU7NFBRI26RAN7GI5POFBB"),
+ *       ),
+ *     )
+ *   ); ===> complex object
+ *
+ *   scValByType(accountAddress) returns "GBBM6BKZPEHWYO3E3YKREDPQXMS4VK35YLNU7NFBRI26RAN7GI5POFBB"
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const scValByType = (scVal: xdr.ScVal): any => {
```

Comment:
Do we want to type it? (I'm guessting it'd look like `String | Json | Array | Error` etc)

Reply:
Yeah, I thought about typing it too but there is this one line from scvLedgerKey types which returns a lot of different types:
![Screenshot 2024-03-28 at 20 04 44](https://github.com/stellar/typescript-wallet-sdk/assets/3228151/2bc77c97-8a07-4c57-b941-68c83c008fca)

Reply:
It's also not possible to do `scVal.value().toString()` to force a string outcome

Any suggestions on how to better type this?

Reply:
Actually, I think I'll remove the explicit `any` type and let TypeScript infer it for us :)